From 710019759bce6bad04553591e70c8755cc17b7b8 Mon Sep 17 00:00:00 2001
From: Robert Nelson <robertcnelson@gmail.com>
Date: Fri, 15 Mar 2013 10:16:03 -0500
Subject: [PATCH] mx51evk: uEnv.txt, bootz, n fixes

Defaults:
#define CONFIG_BOOTDELAY	3

#define CONFIG_CMD_FAT		/* FAT: File System support */
#define CONFIG_CMD_EXT2		/* EXT2: File System support */

/* enable: Generic File System Load Commands */
#define CONFIG_FS_FAT
#define CONFIG_FS_EXT4
#define CONFIG_CMD_FS_GENERIC

#define CONFIG_CMD_BOOTZ		/* bootz: enable zImage support */
#define CONFIG_SUPPORT_RAW_INITRD	/* bootz: enable raw initrd support */

fatload -> load
bootm -> bootz
loaduimage -> loadzimage

Signed-off-by: Robert Nelson <robertcnelson@gmail.com>
---
 include/configs/mx51evk.h |   54 ++++++++++++++++++++++++++++++++-------------
 1 file changed, 39 insertions(+), 15 deletions(-)

diff --git a/include/configs/mx51evk.h b/include/configs/mx51evk.h
index cb3d938..30c7737 100644
--- a/include/configs/mx51evk.h
+++ b/include/configs/mx51evk.h
@@ -89,6 +89,17 @@
 #define CONFIG_CMD_FAT
 #define CONFIG_DOS_PARTITION
 
+#define CONFIG_CMD_FAT		/* FAT: File System support */
+#define CONFIG_CMD_EXT2		/* EXT2: File System support */
+
+/* enable: Generic File System Load Commands */
+#define CONFIG_FS_FAT
+#define CONFIG_FS_EXT4
+#define CONFIG_CMD_FS_GENERIC
+
+#define CONFIG_CMD_BOOTZ		/* bootz: enable zImage support */
+#define CONFIG_SUPPORT_RAW_INITRD	/* bootz: enable raw initrd support */
+
 /*
  * Eth Configs
  */
@@ -145,22 +156,35 @@
 
 #define CONFIG_CMD_DATE
 
-#define CONFIG_BOOTDELAY	1
+#define CONFIG_BOOTDELAY	3
 
 #define CONFIG_ETHPRIME		"FEC0"
 
 #define CONFIG_LOADADDR		0x90800000	/* loadaddr env var */
 
 #define CONFIG_EXTRA_ENV_SETTINGS \
+	"bootenv=uEnv.txt\0" \
+	"bootfile=zImage\0" \
 	"script=boot.scr\0" \
 	"uimage=uImage\0" \
+	"loadaddr=0x90800000\0" \
+	"dtb_addr=0x91FF0000\0" \
+	"dtb_file=imx51-babbage.dtb\0" \
+	"console=ttymxc0,115200n8\0" \
+	"optargs=\0" \
 	"mmcdev=0\0" \
-	"mmcpart=2\0" \
-	"mmcroot=/dev/mmcblk0p3 rw\0" \
-	"mmcrootfstype=ext3 rootwait\0" \
-	"mmcargs=setenv bootargs console=ttymxc0,${baudrate} " \
+	"mmcpart=1\0" \
+	"mmcroot=/dev/mmcblk0p2 ro\0" \
+	"mmcrootfstype=ext4 rootwait\0" \
+	"mmcargs=setenv bootargs console=${console} " \
+		"${optargs} " \
 		"root=${mmcroot} " \
 		"rootfstype=${mmcrootfstype}\0" \
+	"loadbootenv=load mmc ${mmcdev}:${mmcpart} ${loadaddr} ${bootenv}\0" \
+	"importbootenv=echo Importing environment from mmc ...; " \
+		"env import -t ${loadaddr} ${filesize}\0" \
+	"loadzimage=load mmc ${mmcdev}:${mmcpart} ${loadaddr} ${bootfile}\0" \
+	"loaddtb=load mmc ${mmcdev}:${mmcpart} ${dtb_addr} /dtbs/${dtb_file}\0" \
 	"loadbootscript=" \
 		"fatload mmc ${mmcdev}:${mmcpart} ${loadaddr} ${script};\0" \
 	"bootscript=echo Running bootscript from mmc ...; " \
@@ -168,7 +192,7 @@
 	"loaduimage=fatload mmc ${mmcdev}:${mmcpart} ${loadaddr} ${uimage}\0" \
 	"mmcboot=echo Booting from mmc ...; " \
 		"run mmcargs; " \
-		"bootm\0" \
+		"bootz ${loadaddr} - ${dtb_addr}\0 " \
 	"netargs=setenv bootargs console=ttymxc0,${baudrate} " \
 		"root=/dev/nfs " \
 		"ip=dhcp nfsroot=${serverip}:${nfsroot},v3,tcp\0" \
@@ -178,15 +202,15 @@
 
 #define CONFIG_BOOTCOMMAND \
 	"mmc dev ${mmcdev}; if mmc rescan; then " \
-		"if run loadbootscript; then " \
-			"run bootscript; " \
-		"else " \
-			"if run loaduimage; then " \
-				"run mmcboot; " \
-			"else run netboot; " \
-			"fi; " \
-		"fi; " \
-	"else run netboot; fi"
+		"echo SD/MMC found on device ${mmcdev};" \
+		"if run loadbootenv; then " \
+			"run importbootenv;" \
+		"fi;" \
+		"if run loadzimage; then " \
+			"run loaddtb;" \
+			"run mmcboot;" \
+		"fi;" \
+	"fi;"
 
 #define CONFIG_ARP_TIMEOUT	200UL
 
-- 
1.7.10.4

